backend/server/src/services/matchService.ts

```diff
@@ -947,6 +965,38 @@ export class MatchService {
     };
   }
 
+  private async findTeamByPlayerId(
+    playerId: Types.ObjectId,
+    tournamentId: Types.ObjectId
+  ): Promise<{
+    id: Types.ObjectId;
+    name: string;
+    players: Types.ObjectId[];
+  } | null> {
+    const tournament = await TournamentModel.findById(tournamentId).exec();
+
+    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
+    if (tournament == null || tournament.teams == null) {
+      return null;
+    }
+
+    for (const team of tournament.teams) {
+      const playerIds = team.players.map((p) =>
+        p instanceof Types.ObjectId ? p : p.id
+      );
+
+      if (playerIds.some((id) => id.equals(playerId))) {
```

Comment:
I believe this works so no need to fix. Just a note that instead of for loop I would suggest something like:
```
const team = teams.find((team) => {
  team.players.some((p)  => {
    (p instanceof Types.ObjectId ? p : p.id).equals(playerId)
  )
)
if (team) return ...
```